Realm/RLMResults.mm

```diff
@@ -560,6 +592,15 @@ - (id)objectAtIndex:(NSUInteger)index {
     @throw RLMException(@"Index is out of bounds.", @{@"index": @(index)});
 }
 
+- (id)valueForKey:(NSString *)key {
+    RLMResultsValidate(self);
+    return @[];
+}
+
+- (void)setValue:(__unused id)value forKey:(__unused NSString *)key {
+    RLMResultsValidateInWriteTransaction(self);
```

Comment:
It's not really necessary, but I do think it'd be better to have a consistent interface and do the checks here.